petastorm/fs_utils.py

```diff
@@ -33,6 +34,9 @@ def get_dataset_path(parsed_url):
         # s3/gs/gcs filesystem expects paths of the form `bucket/path`
         return parsed_url.netloc + parsed_url.path
 
+    if parsed_url.scheme.lower() in ['file'] and "Windows" in platform.system():
+        return parsed_url.path[1:]
```

Comment:
There is a chance that this fix would break for file urls on network location (as per this [article](https://en.wikipedia.org/wiki/File_URI_scheme#Windows)). Consider the example of the following WIndows url: file://hostname/path/to/the%20file.txt